webdriver-spec.html

```diff
@@ -3138,96 +3148,75 @@
+  <li><p>Return <var>result</var>.
+</ol>
 
-<p>This strategy MUST return all <a href="http://www.w3.org/TR/html5/text-level-semantics.html#the-a-element">"a"</a> elements with visible text exactly and case sensitively equal to the term being searched for.</p>
+<h3>Partial Link Text</h3>
 
-<div class="note">
-      <p>This is equivalent to:</p>
-      <pre>
-            elements = driver.find_elements(by = By.TAG_NAME, value = "a")
-            result = []
-            for element in elements:
-              text = element.text
-              if text == search_term:
-                result.append(element)
-          </pre>
-          <p>Where "search_term" is the link text being searched for, and "result" contains the list of elements to return.</p>
-    </div>
-</section> <!-- /Link Text -->
+<p>The <dfn data-lt="partial link text selector">Partial Link text</dfn> <a>strategy</a>
+  is very similar to the <a>Link Text</a> <a>strategy</a>, but rather than matching
+  the entire string, only a subset needs to match. That is, this MUST return all
+  <a>a elements</a> with rendered text that partially or completely and case
+  sensitively matches the term being searched for.
 
-<section>
-<h2>Partial Link Text</h2>
+<p>To find a <a>WebElement</a> with the <a>Partial Link Text</a> <a>strategy</a>
+  the following steps need to be completed:
 
-<p>Strategy name: partial link text</p>
+  <ol>
+    <li><p>let <var>elements</var> be the result of calling <a>querySelectorAll</a>,
+      with argument <a>a elements</a>, with the <a>context object</a> equal to the
+      <var>start node</var>.
 
-<p>This strategy MUST be supported by all WebDriver implementations.</p>
+    <li><p>Let <var>result</var> be an empty <a>NodeList</a>.
 
-<p>This strategy is very similar to <a href="#link-text">link text</a>, but rather than matching the entire string, only a subset needs to match. That is, this MUST return all <a href="http://www.w3.org/TR/html5/text-level-semantics.html#the-a-element">"a"</a> elements with visible text that partially or completely and case sensitively matches the term being searched for.</p>
+    <li><p>For each <var>element</var> in <var>elements</var>:
+      <ol>
+        <li><p>Let <var>rendered text</var> be
+         the result of <a>getting <code>innerText</code></a>
+         of <var>element</var>.
 
-<div class="note">
-      <p>This is equivalent to:</p>
-      <pre>
-            elements = driver.find_elements(by = By.TAG_NAME, value = "a")
-            result = []
-            for element in elements:
-              text = element.text
-              if text.search(search_term) != -1:
-                result.append(element)
-          </pre>
-          <p>Where "search_term" is the link text being searched for, and "result" contains the list of elements to return.</p>
-    </div>
-</section> <!-- /Partial Link Text -->
+        <li><p>If the result of <a>indexOf</a> on <var>rendered text</var> with argument
```

Comment:
I don’t think we should mandate the use of _indexOf_. Just say it’s a string that must contain something else. Implementations may not be in JS.